File: ABC/127/E.py
import math

dp = [0 for _ in range(2*10**5+1)]
f = math.factorial

def fact(n):
    if n == 0:
        return 1
    if dp[n] != 0:
        return dp[n]
    else:
        fn = f(n)
        dp[n] = fn
        return fn

def nCr(n,r):
    return fact(n) // fact(r) // fact(n-r)

def distancesum(arr, n): 
      
    # sorting the array. 
    arr.sort() 
      
    # for each point, finding  
    # the distance. 
    res = 0
    sum = 0
    for i in range(n): 
        res += (arr[i] * i - sum) 
        sum += arr[i] 
      
    return res 

flatten = lambda x: [z for y in x for z in (flatten(y) if hasattr(y, '__iter__') and not isinstance(y, str) else (y,))]

div = 10**9 + 7

N, M, K = map(int, input().split())
repeat = nCr(N*M-2, K-2)
f_row = flatten([[i for _ in range(M)] for i in range(N)])
f_col = flatten([[i for i in range(M)] for _ in range(N)])

manhsum = distancesum(f_row, len(f_row)) + distancesum(f_col, len(f_col))
print(((repeat % div) * (manhsum % div))%div)

File: ABC/127/E.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>
#include <map>
#include <stack>
#include <queue>
#include <deque>
#include <cstdio>
#include <cmath>
#include <cstdlib>
#include <climits>
#include <cstring>
#define rep(i, begin, end) for (__typeof(end) i = (begin) - ((begin) > (end)); i != (end) - ((begin) > (end)); i += 1 - 2 * ((begin) > (end)))

using namespace std;

const int N_MAX = 1000000;

int x[N_MAX];
int y[N_MAX];
long int div = pow(10, 9) + 7;
long int fact(long int n); 
  
long int nCr(long int n, long int r) 
{
    return fact(n) / fact(r) / fact(n - r); 
} 

long int binomialCoeff(int n, int k) 
{ 
    int res = 1; 
  
    // Since C(n, k) = C(n, n-k) 
    if ( k > n - k ) 
        k = n - k; 
  
    // Calculate value of [n * (n-1) *---* (n-k+1)] / [k * (k-1) *----* 1] 
    for (int i = 0; i < k; ++i) 
    { 
        res *= (n - i); 
        res /= (i + 1); 
    } 
  
    return res; 
} 

// Returns factorial of n 
long int fact(long int n) 
{ 
    long int res = 1; 
    for (long int i = 2; i <= n; i++) 
        res = res * i; 
    return res; 
} 

long int distancesum(int arr[], int n) 
{ 
    // sorting the array. 
    sort(arr, arr + n); 
  
    // for each point, finding the distance. 
    int res = 0, sum = 0; 
    for (int i = 0; i < n; i++) { 
        res += (arr[i] * i - sum); 
        sum += arr[i]; 
    } 
  
    return res; 
} 

long int totaldistancesum(int x[], int y[], int n, int m) 
{ 
    return distancesum(x, n) + distancesum(y, m);
} 

int main(){
  int N, M, K;
  cin >> N >> M >> K;
  int row, col;
  int i=0;
  int j=0;

  rep(row, 0, N){
    rep(col, 0, M){
      x[i++] = row;
    }
  }
  rep(row, 0, N){
    rep(col, 0, M){
      y[j++] = col;
    }
  }
  long int manh = totaldistancesum(x, y, N*M, N*M) % div;
  long int num = binomialCoeff(N*M-2, K-2);
  long int ans = (manh * num) % div;
  cout << ans << endl;
  return 0;
}
